docs/consumer.md

```diff
@@ -1,356 +1,359 @@
+The mock service offers several important features when building your contracts:
 
-In order to handle variable data and make tests more robust, there are a number of matchers available as described below.
+-   It provides a real HTTP server that your code can contact during the test and returns the responses you defined.
+-   You provide the expectations for the requests your code will make, and it asserts the contents of the actual requests made against your expectations.
+-   If a request is made that does not match one you defined, or if a request from your code is missing, it returns an error with details.
 
-### Terms
+## Broker
 
-The `Term` matcher allows you to define a regular expression that the value should match, along with an example value. The pattern is used by Pact for determining the validity of the response, while the example value is returned by Pact in cases where a response needs to be generated.
+The above example showed how to test a single consumer; however, without also testing the provider, the test is incomplete. The Pact Broker is a service that allows you to share and manage your contracts between your consumer and provider tests. It acts as a central repository for your contracts, allowing you to publish contracts from your consumer tests and retrieve them in your provider tests.
 
-This is useful when you need to assert that a value has a particular format, but you are unconcerned about the exact value.
+Once the tests are complete (and successful), the contracts can be uploaded to the Pact Broker. The provider can then download the contracts and run its own tests to ensure it meets the consumer's expectations.
 
-```python
-body = {
-    "id": 123,
-    "reference": Term(r"[A-Z]\d{3,6}-[0-9a-f]{6}", "X1234-456def"),
-    "last_modified": Term(
-        r"\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}Z",
-        "2024-07-20T13:27:03Z",
-    ),
-}
+The Broker CLI is a command-line tool that can be installed through the `pact-python-cli` package, or directly from the [Pact Standalone](https://github.com/pact-foundation/pact-standalone) releases page. It bundles several standalone CLI tools, including the `pact-broker` CLI client.
 
-(
-    pact.given("User 123 exists")
-    .upon_receiving("a request for user 123")
-    .with_request("get", "/user/123")
-    .will_respond_with(200, body=body, headers={
-        "X-Request-ID": Term(
-            r"[a-z]{4}[0-9]{8}-[A-Z]{3}",
-            "abcd1234-EFG",
-        ),
-    })
-)
-
-client = UserClient(pact.uri)
+The general syntax for the CLI is:
 
-with pact:
-    result = client.get_user(123)
-    assert result["id"] == 123
-    assert result["reference"] == "X1234-456def"
-    assert result["last_modified"] == "2024-07-20T13:27:03Z"
+```console
+pact-broker publish \
+    /path/to/pacts/consumer-provider.json \
+    --consumer-app-version 1.0.0 \
+    --tag-with-git-branch \
+    --auto-detect-version-properties
```

Comment:
lets avoid tags if we can, and remove the consumer-app-version, as its preferred there is at least a sha identifier as part of the version, which `auto-detect-version-properties` will pick up